File: pdlearning/src/web-apps/src/libraries/domain-api/src/comment/dtos/create-comment-request.ts
import { CommentNotification } from '../models/comment-notification';
import { EntityCommentType } from '../models/entity-comment-type';

export interface ICreateCommentRequest {
  objectId: string;
  content: string;
  entityCommentType?: EntityCommentType;
  commentNotification?: CommentNotification;
}

export class CreateCommentRequest implements ICreateCommentRequest {
  public objectId: string;
  public content: string;
  public entityCommentType?: EntityCommentType;
  public commentNotification?: CommentNotification;
}

File: pdlearning/src/microservices/course-microservice/Api/Application/RequestDtos/CommentRequest/CreateCommentRequest.cs
using System;
using Microservice.Course.Application.Enums;

namespace Microservice.Course.Application.RequestDtos
{
    public class CreateCommentRequest
    {
        public string Content { get; set; }

        public Guid ObjectId { get; set; }

        public EntityCommentType EntityCommentType { get; set; }

        public CommentNotification? CommentNotification { get; set; }
    }
}
